types/replacestream/index.d.ts

```diff
@@ -0,0 +1,51 @@
+// Type definitions for replacestream 4.0
+// Project: https://github.com/eugeneware/replacestream#readme
+// Definitions by: Piotr Roszatycki <https://github.com/dex4er>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+export interface ReplaceStreamOptions {
+    /**
+     * Sets a limit on the number of times the replacement will be made. This
+     * is forced to one when a regex without the global flag is provided.
+     *
+     * Default: `Infinity`
+     */
+    limit?: number;
+    /**
+     * The text encoding used during search and replace.
+     *
+     * Default: `"utf8"`
+     */
+    encoding?: string;
+    /**
+     * When doing cross-chunk replacing, this sets the maximum length match
+     * that will be supported.
+     *
+     * Default: `100`
+     */
+    maxMatchLen?: number;
+    /**
+     * When doing string match (not relevant for regex matching) whether to do a
+     * case insensitive search.
+     *
+     * Default: `true`
+     */
+    ignoreCase?: boolean;
+    /**
+     * When provided, these flags will be used when creating the search regexes
+     * internally.
+     *
+     * @deprecated as the flags set on the regex provided are no longer mutated
+     * if this is not provided.
+     */
+    regExpOptions?: string;
+}
+
+export type ReplaceStreamFunction = (match: string, p1: string, offset: number, string: string) => string;
+
+export default function ReplaceStream(
```

Comment:
It looks like `require("replacestream)` is itself a function, not an object with a `default` property. So this should be:
```ts
function ReplaceStream(...): any;
export = ReplaceStream;
```